src/changes/applyChanges.ts

```diff
@@ -159,36 +188,77 @@ export function applyAcceptedRejectedChanges(
     }
 
     if (change.dataTracked.status === CHANGE_STATUS.accepted) {
-      // Find the original delete change for this move
-      const originalChange = changeSet.changes.find(
+      // Remove tracking from the moved node (new position)
+      const attrs = {
+        ...node.attrs,
+        dataTracked: getUpdatedDataTracked(node.attrs.dataTracked, change.id),
+      }
+      tr.setNodeMarkup(from, undefined, attrs, node.marks)
+
+      // Find all the original delete changes for this move (there can be many)
+      const originalChanges = changeSet.changes.filter(
         (c) =>
           c.dataTracked.moveNodeId === change.dataTracked.moveNodeId &&
           c.dataTracked.operation === CHANGE_OPERATION.delete
       )
 
-      if (originalChange) {
-        const { pos: originalFrom } = deleteMap.mapResult(originalChange.from)
-        const originalNode = tr.doc.nodeAt(originalFrom)
+      if (originalChanges.length === 0) {
+        log.warn('No original change found for move operation', { change })
+      }
 
-        // Remove tracking from the moved node (new position)
-        const attrs = {
-          ...node.attrs,
-          dataTracked: getUpdatedDataTracked(node.attrs.dataTracked, change.id),
+      originalChanges.forEach((originalChange) => {
+        const { pos: originalFrom, deleted } = deleteMap.mapResult(originalChange.from)
+        if (deleted) {
+          return
         }
-        tr.setNodeMarkup(from, undefined, attrs, node.marks)
+
+        const originalNode = tr.doc.nodeAt(originalFrom)
 
         // Delete the original node (old position)
         if (originalNode) {
           tr.delete(originalFrom, originalFrom + originalNode.nodeSize)
           deleteMap.appendMap(tr.steps[tr.steps.length - 1].getMap())
         }
-      } else {
-        log.warn('No original change found for move operation', { change })
-      }
+      })
     } else if (change.dataTracked.status === CHANGE_STATUS.rejected) {
-      // For rejected moves, delete the moved node (new position)
+      // Collect all moveNodeIds from the moved node and its descendants
+      const moveNodeIdsToRestore = collectMoveNodeIds(node, change.dataTracked.moveNodeId!)
```

Comment:
can you please elaborate a bit on why we need to collect and restore all moveNodeIds?

Reply:
We are collecting all moveNodeIds specifically inside that node to ensure complete restoration in case of sequential indentations. If we didn't restore all, the doc will have some orphan nodes still marked as "deleted" but are no longer part of a valid move operation.

Reply:
I would say, we don't need to guarantee tracking sequential moves. We can revisit that later. Please add this comment of yours here right into the code so it's more clear what's the use case when reading the code. Thanks!